components/tdisplays3/t_display_s3.cpp

```diff
@@ -8,26 +8,132 @@ namespace tdisplays3 {
 
 static const char *const TAG = "TDisplayS3";
 
+#ifdef TDISPLAYS3_USE_ASYNC_IO
+namespace writer_shim {
+  static TaskHandle_t writeFrameTask_ = nullptr;
+  static int16_t x1 = 0;
+  static int16_t x2 = 0;
+  static int16_t y1 = 0;
+  static int16_t y2 = 0;
+  static uint16_t *buffer = nullptr;
+  static void (*callback)(void) = nullptr;
+
+  // A background task used to write the framebuffer to the display
+  static void IRAM_ATTR write_framebuffer_task(void *pv_params) {
+
+    auto tft = (TFT_eSPI *) pv_params;
+
+    while (true) {
+      ulTaskNotifyTake(pdTRUE, portMAX_DELAY);
+
+      if (x1 == 0 && y1 == 0 && x2 == 0 && y2 == 0) {
+        continue;
+      }
+
+      //And push the image
+      tft->pushImage(x1, y1, (x2 - x1) + 1, (y2 - y1) + 1, buffer);
+      
+      //Notify we are done!
+      callback();
+    }
+  }
+
+  static void init(TFT_eSPI *tft) {
+    //The stacksize needed for this driver is about 724 bytes, but we need to be safe, so take some extra just to be sure..
+    //The task is also pinned to CPU core 0, with low priority (just above idle), to decouple writing and rendering.    
+    xTaskCreatePinnedToCore(writer_shim::write_framebuffer_task, "tft_drv", 1024, tft, 1, &writeFrameTask_, 0); //tskNO_AFFINITY
```

Comment:
could you create some (meaningfully) named constants for those magic numbers? It would make the code better readable IMHO.